census/management/commands/google_publish.py

```diff
@@ -16,7 +16,7 @@ class Command(BaseCommand):
     def handle(self, *args, **options):
         # TODO optimize the SQL query
         log.info('starting publish...')
-        for event in Event.objects.all():
+        for event in Event.objects.filter(approval_status=constants.EventApprovalStatus.APPROVED):
```

Comment:
:man_shrugging: If I change it to `.name`, the test fails. So something is inconsistent with using the `.name`.

I was following [this article](https://hackernoon.com/using-enum-as-model-field-choice-in-django-92d8b97aaa63), which doesn't use `.name`. I think as long as we are consistent, it should work. I prefer not using `.name` if possible, since the code is easier to read.

Some investigation shows that `PENDING` vs `PENDING.name` are not the same, so consistency is important.
```
$ pipenv run python manage.py shell
Loading .env environment variables…
Python 3.6.9 (default, Nov 10 2019, 17:30:15)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from census.constants import EventApprovalStatus
>>> EventApprovalStatus.PENDING
<EventApprovalStatus.PENDING: 'Pending'>
>>> EventApprovalStatus.PENDING.name
'PENDING'
>>> str(EventApprovalStatus.PENDING)
'EventApprovalStatus.PENDING'
```

@n8snyder can you shed some light on this?

Reply:
Ah, yes the test is written wrong because we are using `.name` in the models. Apparently shortly after implementing that article I ran into some confusion and broke that consistency. https://github.com/openoakland/dos-census-events/commit/7ab19ade93a00b70d4373ad37fac7f857cd69c95#diff-7fc1bb6d213f0870cd931b3913a75f18

I think we should go back and drop `.name` throughout the code. I'm also curious why Django allows the inconsistency. If the model is defined with the choice 'PENDING', (`PENDING.name`), why doesn't it complain when we use 'EventApprovalStatus.PENDING' to save the model?

Reply:
So are you going to make the global change?  I noticed previously if you don't use .name then EventApprovalStatus.PENDING' is what goes into the database rather than just PENDING.  Is this what we want?

Reply:
I'm suggesting that we make the global change, but looking for input from you and others. If I get a :+1: then I will follow up with the change.

The name in the DB is not so important to me. I want to be able to use the Enums in the code like `EventApprovalStatus.PENDING` (without `.name`). So I am okay with the longer value in the database as a consequence.

Reply:
I agree it would be good for @n8snyder to comment before we make the change.

Reply:
To be honest I haven't used an Enum before with choices, but I'm all having less code 👍

Reply:
I finally got around to looking things over. Because we defined choices as (t.name, t.value), when data gets saved from the form it saves name, such as "PENDING". The reason your test is working is because the test data you create is EventApprovalStatus.APPROVED instead of EventApprovalStatus.APPROVED.name, so the test data does not reflect real data that gets created by the form. I think it is worth refactoring to remove the .name from everything, but probably do a separate branch for that.


As for why django doesn't complain when saving an invalid choice, it is because clean_fields() is not called on save. We would have to override the save method to do that, as seen [here](https://stackoverflow.com/questions/18803112/django-doesnt-call-model-clean-method). It is probably worth doing this change.

Reply:
I will do the enum refactor as part of this PR and open an issue to add clean_fields.

Reply:
Ugg.. it doesn't look like this is going to work. I guess when we don't specify `.name`, the enum object is used. In most cases, it will be cast to string with `str`, but not in all cases. When the form validates, it sees the input `"EventType.WORKSHOP"` but complains it is not a valid choice. This might be a bug in Django.

```
>>> EventForm().fields['event_type'].choices
[('', '---------'), (<EventType.WORKSHOP: 'Workshop'>, 'Workshop'), (<EventType.TABLING: 'Tabling'>, 'Tabling'), (<EventType.CELEBRATION: 'Celebration'>, 'Celebration'), (<EventType.DROPIN: 'Drop-in center'>, 'Drop-in center'), (<EventType.OTHER: 'Other'>, 'Other')]
```

To keep things consistent and avoid ambiguity, I'm thinking we drop the Enums from choices and just use strings/tuples the old school way.